src/util/dstring.h

```diff
@@ -90,6 +90,22 @@ class dstringt final
     return no==0; // string 0 is exactly the empty string
   }
 
+  /// equivalent of as_string().starts_with(s)
+  bool starts_with(const char *s)
+  {
+    for(const char *t = c_str(); *s != 0; s++, t++)
+      if(*t != *s)
```

Comment:
This is out-of-bounds when `this->size()` is less than `strlen(s)`.

Reply:
`c_str` guarantees zero-termination.